types/autocomplete.js.flow

```diff
@@ -1,21 +1,22 @@
 /* @flow */
 
-// One of text or snippet is required.
-// TODO(hansonw): use a union + intersection type
 type atom$AutocompleteSuggestion = {
-  text?: string,
-  snippet?: string,
+  // @NOTE: One of `text` or `snippet` is required.
```

Comment:
While yes only one is required, in theory, they are both optional with a condition of having the other. Therefore, you should leave this as 2 optional properties.

Reply:
In theory, yes. But note that these type definitions are only used for Hydrogen, and we don't need stick to the strict (, which is made for general public usages) definition here.
We all know we only use `text` property for Hydrogen, since no Jupyter kernel offers `snippet` or `signature` completions, and why not use this knowledge ?

If we leave `text` as nullable field, we need additional (and no actually needed) condition checks, which seems to me very weird.

Reply:
In my opinion you should mention this in a comment then. Something like we only use text for Hydrogen but another option is using snippet. I would then be fine with this change and you can ignore the other related flow type comments